### ISSUE TITLE
Sendmail transport is broken in 8.1.3

### ISSUE THREAD

**Issue by zerkms** [owncloud/core#19110]
In the `mailer.php` that's how email is sent:

    $mailer->send($message->getSwiftMessage(), $failedRecipients);

where `$mailer` in an instance of `Swift_SendmailTransport` (that is created with `return \Swift_SendmailTransport::newInstance($binaryPath . ' -bs');`)

The problem is that the transport **must be** started with the `start()` method, otherwise its buffers are not initialized.

It's not obvious why the OC uses the transport directly without `Swift_Mailer`, which handles it properly: https://github.com/swiftmailer/swiftmailer/blob/5.x/lib/classes/Swift/Mailer.php#L78

https://www.dropbox.com/s/7rdogx7r3ct4gmo/screenshot-oc.png?dl=0

On the screenshot you can see that the in/out streams are not initialized, so transport cannot function properly.

**Comment by karlitschek** [comment]
@LukasReschke what do you think?

**Comment by LukasReschke** [comment]
@zerkms Thanks a lot for your bug report.

I tried to initialize the transport there manually as well but this doesn't seem to work reliably:

```
	/**
	 * Returns the sendmail transport
	 *
	 * @return \Swift_SendmailTransport
	 */
	protected function getSendMailInstance() {
		switch ($this->config->getSystemValue('mail_smtpmode', 'sendmail')) {
			case 'qmail':
				$binaryPath = '/var/qmail/bin/sendmail';
				break;
			default:
				$binaryPath = '/usr/sbin/sendmail';
				break;
		}

		$transport = \Swift_SendmailTransport::newInstance($binaryPath . ' -bs');
		$transport->start();
		return $transport;
	}
```

May I ask you whether https://github.com/owncloud/core/pull/19047 would be sufficient for you as well? If you have alternative suggestions I'd greatly appreciate and welcome a Pull Request. :smiley:

**Comment by zerkms** [comment]
@LukasReschke why not use `Swift_Mailer` instance instead? Not sure about portability, but that's what I use for years (my environments are not very different from one another though)

**Comment by LukasReschke** [comment]
> @LukasReschke why not use Swift_Mailer instance instead? Not sure about portability, but that's what I use for years (my environments are not very different from one another though)

So you would propose `return Swift_Mailer::newInstance($transport)`? - Let me test this on some machines and come up with a PR if it works on all.

**Comment by zerkms** [comment]
At least how Swift mailer developers suggest us to do that :-)

http://swiftmailer.org/docs/sending.html

**Comment by MorrisJobke** [comment]
Fixed with #19657
